numpy/f2py/crackfortran.py

```diff
@@ -2704,11 +2733,15 @@ def analyzevars(block):
             if dim and 'dimension' not in vars[n]:
                 vars[n]['dimension'] = []
                 for d in rmbadname([x.strip() for x in markoutercomma(dim).split('@,@')]):
+                    # d is the expression inside the dimension declaration
                     star = '*'
                     if d == ':':
                         star = ':'
-                    if d in params:
-                        d = str(params[d])
+                    if ((d in params) or
+                            (d.find("(") != -1 and d[:d.find("(")] in params)):
+                        # the dimension for this variable depends on a
+                        # previously defined parameter
+                        d = param_parse(d, params)
                     for p in list(params.keys()):
```

Comment:
I think this `for` loop actually belongs in the final `else` block of `parse_param`